lib/parser.js

```diff
@@ -190,13 +190,27 @@ async function validateAndConvertMessage(msg, originalAsyncAPIDocument, fileForm
 
   const schemaFormat = msg.schemaFormat || DEFAULT_SCHEMA_FORMAT;
 
-  await PARSERS[String(schemaFormat)]({
+  //set for the test
+  global.tempPathToPayload = pathToPayload;
+  global.tempSchemaFormat = schemaFormat;
+
+  const parser = PARSERS[String(schemaFormat)];
+
+  if (parser === undefined) {
+    throw new ParserError({
```

Comment:
we are missing `validationErrors` object that among other things contains exact jsonPointer that points to specific part of the document that is broken (containing schema format that is not supported by the parser) an also `startLine` and others

Reply:
Have a look how this is done in custom validators https://github.com/asyncapi/parser-js/blob/master/lib/customValidators.js#L58

Reply:
Also as derberg wrote in my previous comment, please add info which parser should be used if is missed, something like `For avro format, please use @asyncapi/avro-schema-parser` etc. Please make it for officially supported parsers. For custom schema format please don't add info about it :)

Reply:
just to clarify and make it easier for you, an example:

1. you parse AsyncAPI document and one of messages has `schemaFormat: 'application/vnd.apache.avro;version=1.9.0'`
2. You detect no custom schema parser is there, and that it is [one of avro](https://github.com/asyncapi/avro-schema-parser/blob/master/index.js#L14-L19)
3. You throw errors like `you shall not pass, no schema parser available that can parse this schema. The good news is that we have one for Avro, just install @asyncapi/avro-schema-parser and use registerSchemaParser to register it with the current parser`

Now, this may be tricky to achieve. Avro mime types supported by avro schema parser are defined in avro-schema-parser, not in parser-js, so it can be tricky to come up with some good idea for it. Let us know if ya needs some help figuring that out

Reply:
just to inform you that during this period due to the gsoc and the university exams i dont have time and i will not be able to add this feature. Is this change urgent right now?

Reply:
No. We know about university and gsoc, don't worry, we are very happy that you find free time and wanna help us :) You can end it in the next weeks (or maybe also months :P), or if we decide that it's urgent, someone handle that for you, ok? :)

Reply:
@magicmatatjahu as the PR actually solves the issue (maybe not with the extended features you want) does it make sense to split up the issue? So this PR can be merged as is now? Or do you want to have this PR stay open until someone picks it up?

Reply:
@jonaslagoni Łukasz will be back on Monday, so we will make review  again and then decide, merge or not :)